components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromOpenAPISpec.java

```diff
@@ -126,6 +134,11 @@
                                     + "' is invalid");
                         }
                     }
+
+                    if (isGlobalParameterDefined && !isHttpVerbDefined) {
+                        handleException("Provided for resource '" + uriTempVal + "' has global parameters without " +
```

Comment:
Do we need the verb 'for'

Reply:
The purpose of this is to prevent a swagger which has global parameters without a corresponding HTTP verb(This would be an invalid swagger). This check was suggested by Thilini since the swagger parser shows this as a warning instead of an error so APIM may try to process such a swagger